src/util/cmdline_definition.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************\
+
+Module: Command line parsing
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+// \file Command line options definition
+
+#ifndef CPROVER_UTIL_CMDLINE_DEFINITION_H
+#define CPROVER_UTIL_CMDLINE_DEFINITION_H
+
+#include <list>
+#include <string>
+#include <vector>
+
+#include "optional.h"
+
+class json_arrayt;
+class xmlt;
+
+struct cmdline_optiont
+{
+  struct argumentt
+  {
+    argumentt(std::string name, std::string type);
+
+    std::string name;
+    std::string type;
+  };
+
+  typedef std::string deprecatedt;
+
+  cmdline_optiont(
+    std::string name,
+    optionalt<argumentt> argument,
```

Comment:
I don't think we should use `optionalt` here (or below, for `deprecated`), just have multiple constructors with varying numbers of parameters. The `optionalt` class members of course still remain in place.

Reply:
Yes, although `{}` flows quite intuitively in the initialiser list...